### PR TITLE
Properly truncate incoming messages from rpi

### PR DESCRIPTION
**Pull request by mgalardini**
The input string is now truncated each time a full message has
arrived; previous behavior was the gradual increase in size of
`inputString`, until it was cleared, which might lead to truncation
of long messages and dropping of commands.

This change has been tested on all arduinos except the `PD` one. The
`STIR` arduino still seems to drop a command every ~10, so it
might need a few more tests to remove all possible corner cases, which
may be due to the `Serial1` messages overlapping.

### CODE REVIEW

File: SAMD21/Rev B/RS485_OD/RS485_OD.ino [hunk 1]
```diff
@@ -202,7 +205,7 @@ void read_MuxShield() {
   
   for (int h=0; h<(PDtimes_averaged); h++){
     mux_total = mux_total + readMux(active_vial);
-    serialEvent(1);
```
[the remark this answers is not in the document]
Comment (mgalardini) [reply SAMD21/Rev B/RS485_OD/RS485_OD.ino]: That's because I changed the `serialEvent` function to not be based on waiting a set amount of time but until either the serial is done sending messages or a `!` char is found.

File: SAMD21/Rev B/RS485_PD/RS485_PD.ino [hunk 2]
```diff
@@ -46,8 +46,8 @@ void setup() {
   analogReadResolution(16);
   Serial1.begin(9600);
   SerialUSB.begin(9600);
-  // reserve 1000 bytes for the inputString:
```
[the remark this answers is not in the document]
Comment (mgalardini) [reply SAMD21/Rev B/RS485_PD/RS485_PD.ino]: I guess it avoids overflows when a very large command is found (biggest possible is something like a large chemostat dilution once a day on all vials).

File: SAMD21/Rev B/RS485_STIR/RS485_STIR.ino [hunk 3]
```diff
@@ -27,19 +27,21 @@ void setup()
   SerialUSB.begin(9600);
   Serial1.begin(9600);
   // reserve 200 bytes for the inputString:
-  inputString.reserve(1000);
-  while (!Serial1);
+  inputString.reserve(2000);
```
[the remark this answers is not in the document]
Comment (mgalardini) [reply SAMD21/Rev B/RS485_STIR/RS485_STIR.ino]: See above, mostly a fail-safe mechanism against very long messages (doesn't apply for stir maybe)

File: SAMD21/Rev B/RS485_STIR/RS485_STIR.ino [hunk 4]
```diff
@@ -27,19 +27,21 @@ void setup()
...
   pinMode(12, OUTPUT);
   digitalWrite(12, LOW);
-  
+  while (!Serial1);
```
[the remark this answers is not in the document]
Comment (mgalardini) [reply SAMD21/Rev B/RS485_STIR/RS485_STIR.ino]: I don't think so, I just wanted to be consistent across arduinos

File: SAMD21/Rev B/RS485_STIR/RS485_STIR.ino [hunk 5]
```diff
@@ -27,19 +27,21 @@ void setup()
...
 }
 
 
 void loop() {
-  serialEvent(1);
```
[the remark this answers is not in the document]
Comment (mgalardini) [reply SAMD21/Rev B/RS485_STIR/RS485_STIR.ino]: See above
